src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1524,6 +1526,66 @@ public void searchStudentsInCourse() throws Exception {
         assertThat(tutors).isEmpty();
     }
 
+    public void searchUsersInCourse() throws Exception {
+        var course = database.createCourse();
+        // Test 1: search for all (no login or name) tutors and editors (tutors includes also editors)
+        var resultTest1 = searchUsersTest(course, List.of("tutors"), Optional.empty(), numberOfTutors + numberOfEditors, true);
+        assertThat(resultTest1.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(numberOfEditors);
+        assertThat(resultTest1.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant).collect(Collectors.toList())).hasSize(numberOfTutors);
+
+        // Test 2: search for all (no login or name) instructors
+        var resultTest2 = searchUsersTest(course, List.of("instructors"), Optional.empty(), numberOfInstructors, true);
+        assertThat(resultTest2.stream().filter(UserPublicInfoDTO::getIsInstructor).collect(Collectors.toList())).hasSize(numberOfInstructors);
+
+        // Test 3: Try to search for all students (should fail)
+        searchUsersTest(course, List.of("students"), Optional.empty(), 0, false);
+
+        // Test 4: Try to search for all students with a too short search term (at least 3 as students are included) (should fail)
+        searchUsersTest(course, List.of("students"), Optional.of("st"), 0, false);
+
+        // Test 4: Try to search for students with a long enough search term (at least 3 as students are included)
+        // Note: -1 as student1 is the requesting user and will not be returned
+        var resultTest4 = searchUsersTest(course, List.of("students"), Optional.of(userPrefix + "student"), numberOfStudents - 1, true);
+        assertThat(resultTest4.stream().filter(UserPublicInfoDTO::getIsStudent).collect(Collectors.toList())).hasSize(numberOfStudents - 1);
+
+        // Test 5: Try to search for all tutors, editors and instructors
+        var resultTest5 = searchUsersTest(course, List.of("tutors", "instructors"), Optional.empty(), numberOfTutors + numberOfEditors + numberOfInstructors, true);
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(numberOfEditors);
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant).collect(Collectors.toList())).hasSize(numberOfTutors);
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsInstructor).collect(Collectors.toList())).hasSize(numberOfInstructors);
+
+        // Test 6: Try to search for all tutors, editors and instructors with search term
+        var resultTest6 = searchUsersTest(course, List.of("tutors", "instructors"), Optional.of(userPrefix + "tutor"), numberOfTutors, true);
+        assertThat(resultTest6.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(0);
```

Comment:
I think there is also an explicit check for emptiness (isEmpty)

Reply:
Changed it in 2ce687d41045b28e8c895f3a76ac91d725bc309a